File: ocaml/Fizz2.ml
let rec range a b =
    if a > b then []
    else a :: range (a+1) b

let fizzbuzz x =
  match (x mod 15) with
    0 -> "FizzBuzz"
  | 3 -> "Fizz"
  | 5 -> "Buzz"
  | 6 -> "Fizz"
  | 9 -> "Fizz"
  | 10 -> "Buzz"
  | _  -> string_of_int x

let () =
  List.iter (fun x -> (Printf.printf "%s\n" (fizzbuzz x))) (range 1 100)

File: scala/Fizz2.scala
object Fizz2 {

	def FizzBuzz(i: Int) = (i % 3, i % 5) match {
		case (0, 0) => "FizzBuzz"
		case (0, _) => "Fizz"
		case (_, 0) => "Buzz"
		case (_, _) => i
	}

	def main(args: Array[String]){
		(1 to 100).map(FizzBuzz).foreach(println)
	}
}